auth/server.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"encoding/json"
+	"net/http"
+
+	utils "github.com/fabric8-analytics/cli-tools/utils"
+	"github.com/rs/zerolog/log"
+)
+
+// RequestServerType is a argtype of RequestServer func
+type RequestServerType struct {
+	SynkToken       string
+	UserID          string
+	Host            string
+	ThreeScaleToken string
+	Debug           bool
+}
+
+// Payload is format for auth requests to Server
+type Payload struct {
+	utils.GenericPayload
+	SnykToken string `json:"snyk_api_token,omitempty"`
+	UUID      string `json:"user_id,omitempty"`
+}
+
+// UserResponse is format for Response from Server
+type UserResponse struct {
+	UUID string `json:"user_id,omitempty"`
+}
+
+// RequestServer is auth request to CRDA server
+func RequestServer(requestParams RequestServerType) string {
+	log.Debug().Msgf("Executing Request Server.")
+	var payload Payload
+	var resData UserResponse
+	requestData := utils.HTTPRequestType{
+		Payload:         payload,
+		Method:          http.MethodPost,
+		Endpoint:        utils.APIUsers,
+		ThreeScaleToken: requestParams.ThreeScaleToken,
+		Host:            requestParams.Host,
+		Debug:           requestParams.Debug,
+	}
+
+	if requestParams.UserID == "" {
+		// If userID is not present, get one from server.
+		apiResponse := utils.HTTPRequest(requestData)
+		resData = validateResponse(apiResponse, requestData.Debug)
+
+		requestParams.UserID = resData.UUID
+
+	}
+	if requestParams.SynkToken != "" {
+		// If Snyk Token is present in input.
+		payload = Payload{
+			UUID:      requestParams.UserID,
+			SnykToken: requestParams.SynkToken,
+		}
+		requestData.Payload = payload
+		requestData.Method = http.MethodPut
+		resData = validateResponse(utils.HTTPRequest(requestData), requestData.Debug)
+	}
+	log.Debug().Msgf("Successfully executed RequestServer.")
+	return requestParams.UserID
+}
+
+// validateResponse validates API Response and Generate Auth Specific API Errors, if required.
+func validateResponse(apiResponse *http.Response, debug bool) UserResponse {
+	log.Debug().Msgf("Executing validateResponse.")
+	var body map[string]string
```

Comment:
Why a map is used instead of defining a corresponding struct?

Reply:
Ack